src/test/java/com/fasterxml/uuid/ext/LockedFileTest.java

```diff
@@ -0,0 +1,261 @@
+package com.fasterxml.uuid.ext;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static com.fasterxml.uuid.ext.LockedFile.READ_ERROR;
+import static org.junit.Assert.*;
+
+public class LockedFileTest
+{
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Test
+    public void constructor_givenNull_shouldThrowNullPointerException() throws IOException {
+        try {
+            new LockedFile(null);
+            fail("This should have thrown a null pointer exception");
+        } catch (NullPointerException nullPointerException) {
+            assertNull(nullPointerException.getMessage());
+        }
+    }
+
+    @Test
+    public void constructor_givenEmptyFile_shouldLeaveFileAsIs() throws IOException {
+        // given
+        File emptyFile = temporaryFolder.newFile();
+
+        // when
+        new LockedFile(emptyFile);
+
+        // then
+        assertTrue(emptyFile.exists());
+        assertTrue(emptyFile.canRead());
+        assertTrue(emptyFile.canWrite());
+    }
+
+    @Test
+    public void constructor_givenNonExistentFile_shouldCreateANewFile() throws IOException {
+        // given
+        File blankFile = temporaryFolder.newFile();
+        File nonExistentFile = new File(blankFile + ".nonexistent");
+
+        if (Files.exists(nonExistentFile.toPath())) {
+            fail("temp file should not exist");
+        }
+
+        // when
+        new LockedFile(nonExistentFile);
+
+        // then - the nonexistent file now exists?
+        assertTrue(Files.exists(nonExistentFile.toPath()));
+        assertTrue(nonExistentFile.canRead());
+        assertTrue(nonExistentFile.canWrite());
+    }
```

Comment:
This was the really cool use case where if you give it a nonexistent file, the LockedFile class will go ahead and create the file. Pretty neat ey?